### PR TITLE
ci: migrate lint job from CircleCI to GHA

### PR DESCRIPTION
**Pull request by josephperrott**

### CODE REVIEW

File: .github/workflows/ci.yml [hunk 1]
```diff
@@ -0,0 +1,53 @@
...
+        run: yarn -s ng-dev pullapprove verify
+      - name: Validate angular robot configuration
+        run: yarn -s ng-dev ngbot verify
+      - name: Confirm code builds with typescript as expected
+        run: yarn -s check-tooling-setup
+      - name: Check commit message
+        if: github.event_name == 'pull_request'
+        run: yarn ng-dev commit-message validate-range ${{ github.event.pull_request.base.sha }} ${{ github.event.pull_request.head.sha }}
+      - name: Check code format
+        if: github.event_name == 'pull_request'
```
Comment (devversion) [inline .github/workflows/ci.yml:52]: Optional nit: Add comments on why we don't check format/commit message for push builds

**Review by devversion — APPROVED** [review]

**Comment by josephperrott** [comment]
Caretaker Note: this is ready to merge and subsequent PRs won't have the `ci/circleci: lint` job.

**Comment by josephperrott** [comment]
This PR was merged into the repository by commit 6c76a83e3110d3a671c834357441e63cb032f6b4.

File: .github/workflows/ci.yml [hunk 2]
```diff
@@ -0,0 +1,57 @@
...
+    steps:
+      - name: Initialize environment
+        uses: angular/dev-infra/github-actions/npm/checkout-and-setup-node@852c3d1898775ec9fbc87213e519f16e78805b57
+        with:
+          cache-node-modules: true
+          node-module-directories: |
+            ./node_modules
+            ./aio/node_modules
+      - name: Install node modules
+        run: yarn install --immutable
```
Comment (devversion) [inline .github/workflows/ci.yml:33]: is --immutable right?

Comment (josephperrott) [reply .github/workflows/ci.yml:33]: Created follow up PR #51012

**Pull request closed without merging**